### DISCUSSION TITLE
Wireless support?

### DISCUSSION THREAD

**Discussion by terencechow** [JellyTitan/Sofle-Pico#36] [Q&A] [answered]
Hi,

I wanted to make a sofle v2 but I realized if I use the cable, my clumsy self would eventually pull the cable out and short one of the sides. Therefore I wanted to build a wireless one. I stumbled on this one because I was looking at the pico w. I don't see any indication here that this supports Pico W and the product roadmap indicates a wireless variant is unclaimed. I just wanted to confirm that this is still the case? Suppose I did get a Pico W, would there be major differences to get it to work? What level of tinkering would it take to get this to work? For reference, this is my first keyboard I would be building.

**Comment by JellyTitan** [comment]
Great question! I've never dabbled directly in wireless before, I only picked up a few tidbits.

I think QMK wireless support isn't strong, so most folks go with ZMK or KMK.  Sofle Pico would probably need to be ported to ZMK or KMK for better wireless support. I've only used QMK, so I can't speak to the level of effort required to port.

On the hardware side, I think bluetooth and nice!nano are the most common split wireless implementations. The Nice! has the same form factor as the ProMicro. All other Sofle's use the ProMicro form factor, so I think the path of least resistance for getting a Sofle wireless build running would be to adapt the Sofle V1/V2/RGB/Choc. (It looks like somebody already created a [wireless version of the Sofle Choc](https://github.com/db-ok/SofleChocWireless) and ported to ZMK).

It looks like the [Pico W does technically support bluetooth](https://www.raspberrypi.com/news/new-functionality-bluetooth-for-pico-w/) - but I'm not sure that would work with ZMK/KMK. I'm out of my league when it comes to BLE driver stuff.

When I add new features, I'll usually dig around in QMK to find an open-source keyboard that has a feature that I want, and then use that for reference.

If you can [find a Pico W based wireless board](https://golem.hu/boards/), then I don't think adapting the Sofle Pico would be exceptionally difficult, but if there's nothing to crib from, it'll be a heavy lift.

If this is your first build, I recommend going with something standard that has great documentation. There's plenty of unknown unknowns to uncover along the way.

**Comment by JellyTitan** [comment, marked as answer]
Actually - now that I think about it a bit more - maybe this is not a great idea? 
The Pico adds more processing power for OLED and RGB animations - but a wireless version wouldn't have either RGB or OLED? 
In that case, the only real advantage of a wireless Sofle Pico would be the cost of the Pico W vs the Nice!Nano.
The low price point could be an attractive feature - maybe a wireless [cheapino](https://github.com/tompi/cheapino) competitor?

**Reply by terencechow** [reply]
Thanks for the clarification. I'll take your advice and look for a simpler first keyboard. Great project btw!
